soh/soh/Enhancements/game-interactor/GameInteractionEffect.cpp

```diff
@@ -0,0 +1,512 @@
+#include "GameInteractionEffect.h"
+#include "GameInteractor.h"
+#include <libultraship/bridge.h>
+
+extern "C" {
+#include <z64.h>
+#include "variables.h"
+#include "functions.h"
+#include "macros.h"
+extern PlayState* gPlayState;
+}
+
+uint32_t GameInteractor_NoUIActive;
+uint32_t GameInteractor_GiantLinkActive;
+uint32_t GameInteractor_MinishLinkActive;
+uint32_t GameInteractor_PaperLinkActive;
+uint32_t GameInteractor_InvisibleLinkActive;
+uint32_t GameInteractor_ResetLinkScale;
+uint32_t GameInteractor_OneHitKOActive;
+uint32_t GameInteractor_PacifistModeActive;
+uint32_t GameInteractor_DisableZTargetingActive;
+uint32_t GameInteractor_ReverseControlsActive;
+int32_t GameInteractor_DefenseModifier;
+int32_t GameInteractor_RunSpeedModifier;
+uint32_t GameInteractor_GravityLevel = GRAVITY_LEVEL_NORMAL;
```

Comment:
not sure exactly what it'd take to make this happen, and i touched on this in a previous comment, but it'd be for these to be getter/setter methods and have these vars live on the class instead